pkg/controllers/cluster/cluster_controller.go

```diff
@@ -161,14 +307,36 @@ func (r *RemoteClusterReconciler) EnsureDeleted(ctx context.Context, resource li
 		return ctrl.Result{}, lifecycle.Failed, err
 	}
 
-	// deleting the cluster role binding in the remote cluster will delete the greenhouse service account in managed namespace
-	// due to owner reference
-	if err := deleteClusterRoleBindingInRemoteCluster(ctx, remoteClient); err != nil {
+	// deleting the cluster role binding in the remote cluster will delete
+	// greenhouse service account and namespace due to owner reference
+	if err := r.deleteClusterRoleBindingInRemoteCluster(ctx, remoteClient); err != nil {
 		return ctrl.Result{}, lifecycle.Failed, err
 	}
 	return ctrl.Result{}, lifecycle.Success, nil
 }
 
+// deleteClusterRoleBindingInRemoteCluster - deletes the cluster role binding in the remote cluster
+func (r *RemoteClusterReconciler) deleteClusterRoleBindingInRemoteCluster(ctx context.Context, k8sClient client.Client) error {
+	crb := &rbacv1.ClusterRoleBinding{}
+	err := k8sClient.Get(ctx, client.ObjectKey{Name: utils.ServiceAccountName}, crb)
+	if err != nil {
+		ctrl.LoggerFrom(ctx).V(5).Error(err, "potential err getting clusterRoleBinding")
+		if apierrors.IsUnauthorized(err) || apierrors.IsNotFound(err) || apierrors.IsForbidden(err) {
+			return nil
+		}
+		return err
+	}
```

Comment:
Since 401/403/404 are ignored, the logging could move below to only log errors that are returned? (same for the err handling below) 
```suggestion
		if apierrors.IsUnauthorized(err) || apierrors.IsNotFound(err) || apierrors.IsForbidden(err) {
			return nil
		}
		ctrl.LoggerFrom(ctx).V(5).Error(err, "getting clusterRoleBinding failed")
		return err
	}
```